lightning/src/ln/channel.rs

```diff
@@ -5078,6 +5092,7 @@ where
 	) -> u64 {
 		if funding.get_channel_type().supports_anchor_zero_fee_commitments() {
 			debug_assert_eq!(self.feerate_per_kw, 0);
+			debug_assert!(fee_spike_buffer_htlc.is_none());
```

Comment:
Wondering whether it's worth having this assertion (which makes it necessary to check channel type before calling fee functions). The alternative would be to just ignore the parameters completely for zero fee channels (even though `Some` `fee_spike_buffer_htlc` doesn't make sense for the type).